code/processes/gateway.q

```
@@ -462,6 +469,7 @@ syncexecj:{[query;servertype;joinfunction]
  handles:(exec serverid!handle from tab)first each (exec serverid from tab) inter/: serverids;
  setserverstate[handles;1b];
  start:.z.p;
+ $[checkQuery[query;servertype];:"ERROR: Date not provided (.i.e Select from [table] where date=[date])";::];
```

Again use `if`

---

this code is repeated - perhaps on case of error, the `checkQuery` function should return the error signal, to avoid repeating the error message everywhere it's called?